File: pgy_parent/pyg_user_interface/src/main/java/com/pinyougou/user/service/TbUserService.java
package com.pinyougou.user.service;

import com.pinyougou.pojo.TbUser;
import domaincommon.Result;

/**
 * Create with www.dezhe.com
 *
 * @Author 德哲
 * @Date 2018/7/27 21:13
 */
public interface TbUserService {
    void sendCode(String phone) throws Exception;

    void saveUser(TbUser user);

    void checkCode(String phone,String code);
}

File: pgy_parent/pyg_user_web/src/main/webapp/js/service/tbUserService.js
app.service("tbUserService",function ($http) {

    this.showUserNameInit = function () {
        return $http.get("../user/showUserName");
    };
    this.sendCode = function (phone) {
        return $http.post("/user/getCode/"+phone);
    }

    this.userSave = function (user,code) {
        return $http.post("/user/saveUser/"+code,user);
    }

});